integration-tests-scala/src/test/scala/tech/beshu/ror/integration/proxy/ProxyTestSupport.scala

```diff
@@ -0,0 +1,67 @@
+/*
+ *    This file is part of ReadonlyREST.
+ *
+ *    ReadonlyREST is free software: you can redistribute it and/or modify
+ *    it under the terms of the GNU General Public License as published by
+ *    the Free Software Foundation, either version 3 of the License, or
+ *    (at your option) any later version.
+ *
+ *    ReadonlyREST is distributed in the hope that it will be useful,
+ *    but WITHOUT ANY WARRANTY; without even the implied warranty of
+ *    MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ *    GNU General Public License for more details.
+ *
+ *    You should have received a copy of the GNU General Public License
+ *    along with ReadonlyREST.  If not, see http://www.gnu.org/licenses/
+ */
+package tech.beshu.ror.integration.proxy
+
+import better.files._
+import cats.effect.{ContextShift, IO}
+import com.dimafeng.testcontainers.ForAllTestContainer
+import org.scalatest.{BeforeAndAfterAll, Suite}
+import tech.beshu.ror.proxy.{RorProxy, RorProxyApp}
+import tech.beshu.ror.utils.containers.ContainerUtils
+import tech.beshu.ror.utils.containers.generic.{CallingProxy, EsWithoutRorPluginContainerCreator, TargetEsContainer}
+
+import scala.concurrent.ExecutionContext
+
+trait ProxyTestSupport
+  extends BeforeAndAfterAll
+    with ForAllTestContainer
+    with CallingProxy
+    with EsWithoutRorPluginContainerCreator {
+  this: Suite with TargetEsContainer =>
+
+  def rorConfigFileName: String
+
+  private var closeHandler: Option[RorProxy.CloseHandler] = None
+  override val proxyPort = 5000
+
+  override def afterStart(): Unit = {
+    super.afterStart()
+    closeHandler = createApp()
+      .start
+      .unsafeRunSync()
+      .toOption
+  }
+
+  override protected def afterAll(): Unit = {
+    super.afterAll()
+    closeHandler.getOrElse(throw new Exception("Could not start test proxy instance"))().unsafeRunSync()
+  }
+
+  //TODO: Create proxy instance dynamically based on 'esModule' property. Now it's fixed on es74-cloud
+  private def createApp(): RorProxy = new RorProxyApp {
```

Comment:
Now we create `RorProxyApp` from `es74x-cloud` and we have direct dependency `testCompile project(':es74x-cloud')` as well, but that's not what we want, right? We would like to somehow create proxy instance based on `esModule` property and have no dependencies to cloud modules. Is it possible?

Reply:
Or do not create proxy tests in `integration-tests-scala` module but in cloud modules: `es74x-cloud` where `RorProxyApp` would be provided